superset-frontend/src/explore/components/ExploreChartHeader/index.jsx

```diff
@@ -248,134 +226,138 @@ export class ExploreChartHeader extends React.PureComponent {
       ),
     );
     return permissions[0].length > 0;
-  }
+  };
 
-  render() {
-    const { user, form_data: formData, slice } = this.props;
-    const {
-      chartStatus,
-      chartUpdateEndTime,
-      chartUpdateStartTime,
-      latestQueryFormData,
-      queriesResponse,
-    } = this.props.chart;
-    // TODO: when will get appropriate design for multi queries use all results and not first only
-    const queryResponse = queriesResponse?.[0];
-    const chartFinished = ['failed', 'rendered', 'success'].includes(
-      this.props.chart.chartStatus,
-    );
-    return (
-      <StyledHeader id="slice-header" className="panel-title-large">
-        <div className="title-panel">
-          {slice?.certified_by && (
-            <>
-              <CertifiedBadge
-                certifiedBy={slice.certified_by}
-                details={slice.certification_details}
-              />{' '}
-            </>
-          )}
-          <EditableTitle
-            title={this.getSliceName()}
-            canEdit={
-              !this.props.slice ||
-              this.props.can_overwrite ||
-              (this.props.slice?.owners || []).includes(
-                this.props?.user?.userId,
-              )
-            }
-            onSaveTitle={this.props.actions.updateChartTitle}
-          />
+  useEffect(() => {
+    const { userId } = user;
+    const { id } = chart;
 
-          {this.props.slice && (
-            <StyledButtons>
-              {user.userId && (
-                <FaveStar
-                  itemId={this.props.slice.slice_id}
-                  fetchFaveStar={this.props.actions.fetchFaveStar}
-                  saveFaveStar={this.props.actions.saveFaveStar}
-                  isStarred={this.props.isStarred}
-                  showTooltip
-                />
-              )}
-              {this.state.isPropertiesModalOpen && (
-                <PropertiesModal
-                  show={this.state.isPropertiesModalOpen}
-                  onHide={this.closePropertiesModal}
-                  onSave={this.props.sliceUpdated}
-                  slice={this.props.slice}
-                />
-              )}
-              <Tooltip
-                id="edit-desc-tooltip"
-                title={t('Edit chart properties')}
+    if (canAddReports()) {
+      fetchUISpecificReport(userId, 'chart_id', 'charts', id);
+    }
+    if (dashboardId) {
+      fetchChartDashboardData();
+    }
+  }, []);
```

Comment:
It appears that some dependencies might be missing here

Reply:
thanks! I've added dependencies.

Reply:
actually this form of a useEffect emulated a ComponentWillMount which is what the original component was using.